packages/lib-utils/src/k8s/hooks/useK8sWatchResource.test.ts

```diff
@@ -0,0 +1,171 @@
+const useK8sModelMock = jest.mocked(useK8sModel, false);
+const getWatchDataMock = jest.mocked(getWatchData, false);
+const getReduxDataMock = jest.mocked(getReduxData, false);
+
+describe('useK8sWatchResource', () => {
+  beforeEach(() => {
+    jest.resetModules();
+
+    getWatchDataMock.mockReturnValue(null);
+    useSelectorMock.mockReturnValue(null); // get resourceK8s
+    useModelsLoadedMock.mockReturnValue(true);
+    useK8sModelMock.mockReturnValue([undefinedModelMock, false] as [K8sModelCommon, boolean]);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test('should support null as an input value', () => {
+    useDeepCompareMemoizeMock.mockReturnValue({ kind: '__not-a-value__' });
+
+    const { result } = renderHook(() => useK8sWatchResource(null));
+    const [data, loaded, error] = result.current;
+
+    expect(data).toBeUndefined();
+    expect(loaded).toEqual(true);
+    expect(error).toBeUndefined();
+  });
+
+  test('should return "loaded" as false when models have not loaded', () => {
+    const haveModelsLoaded = false;
+    useModelsLoadedMock.mockReturnValue(haveModelsLoaded);
+    useDeepCompareMemoizeMock.mockReturnValue(watchedResourceMock);
+
+    const { result } = renderHook(() => useK8sWatchResource(watchedResourceMock));
+    const [data, loaded, error] = result.current;
+
+    expect(data).toMatchObject({});
+    expect(loaded).toEqual(false);
+    expect(error).toBeUndefined();
+  });
+
+  test('should return specific error if the model for the watched resource does not exist', () => {
+    useDeepCompareMemoizeMock.mockReturnValue(watchedResourceMock);
+
+    const { result } = renderHook(() => useK8sWatchResource(watchedResourceMock));
+    const [data, loaded, error] = result.current;
+
+    expect(data).toMatchObject({});
+    expect(loaded).toEqual(true);
+    expect((error as NoModelError).message).toEqual('Model does not exist');
+  });
+
+  test('should return data for the watched resource', () => {
+    const mockWatchData: WatchData = {
+      id: 'appstudio.redhat.com~v1alpha1~Application---{"ns":"test-ns","name":"test"}',
+      action: jest.fn(),
+    };
+    const payload = {
+      data: resourceDataMock,
+      loaded: true,
+      loadError: '',
+    };
+    const reduxIdPayload: ImmutableMap<string, unknown> = ImmutableMap(payload);
+    useSelectorMock.mockReturnValue(reduxIdPayload); // get resourceK8s
+    getReduxDataMock.mockReturnValue(resourceDataMock);
+    useDeepCompareMemoizeMock.mockReturnValue(watchedResourceMock);
+    useK8sModelMock.mockReturnValue([resourceModelMock, false] as [K8sModelCommon, boolean]);
+    getWatchDataMock.mockReturnValue(mockWatchData);
+
+    const { result } = renderHook(() => useK8sWatchResource(watchedResourceMock));
+    const [data, loaded, error] = result.current;
+
+    expect(useSelectorMock).toHaveBeenCalled();
+    const resourceData = data as K8sResourceCommon;
+
+    expect(resourceData).toEqual(
```

Comment:
So you should be able to replace this entire expect (starting on `151`) with:

```suggestion
    expect(data as K8sResourceCommon).toMatchObject(resourceDataMock);
```

Reply:
I also confirmed this does not effect the coverage and will be a bit more dynamic if we add/remove fields.

Reply:
That's a good point, @florkbr. Thanks! Addressed 👍